keeper/msg_server_vaults.go

```diff
@@ -433,3 +437,73 @@ func (k *Keeper) ToUserVaultPositionModuleAccount(address string, vaultType vaul
 		return vaults.StakedVaultName
 	}
 }
+
+// VaultsEndProgram handles the logic to end the vaults program, unlocking all user positions.
+func (k *Keeper) VaultsEndProgram(ctx context.Context) error {
```

Comment:
nit, but just wondering why is this exported if its only used in the keeper?
I have gotten used to having everything private by default and only export if required, what do you guys think of that? 😇 it's another little helper to immediately understand package scope while reading the code without having to rely on the LSP IMO
```suggestion
func (k *Keeper) vaultsEndProgram(ctx context.Context) error {
```